src/test/unit/bqrc_kalman_unittest.cc

```diff
@@ -0,0 +1,197 @@
+
+#include <stdint.h>
+#include <stdio.h>
+#include <time.h>
+#include <math.h>
+#include "gtest/gtest.h"
+
+extern "C" {
+    #include "common/maths.h"
+    #include "common/filter.h"
+}
+
+#define M_PI_FLOAT 3.14159265358979323846f
+#define SAMPLE_SET_SIZE 500000
+#define PASSING_CORRELATION 0.9999f
+
+
+double corr(double *x, double *y, int startIndex, int n)
+{
+    double sumX = 0;
+    double sumY = 0;
+    double sumXY = 0;
+    double squareSumX = 0;
+    double squareSumY = 0;
+
+    for (int i = startIndex; i <= startIndex + n; i++) {
```

Comment:
This will loop n+1 times ... is that expected?

Reply:
Yes, I think so, there's a `- 1` where it's used.

Reply:
It's confusing for anyone reading it ...

Reply:
I'll make amendments, that why we pushed this branch. Curtis has no time to work on it now